src/_pytest/recwarn.py

```diff
@@ -76,7 +113,11 @@ def warns(expected_warning, *args, match=None, **kwargs):
     """
     __tracebackhide__ = True
     if not args:
-        return WarningsChecker(expected_warning, match_expr=match, **kwargs)
+        # Type ignore because **kwargs is passed intentionally to trigger
+        # a TypeError if provided.
+        return WarningsChecker(  # type: ignore
+            expected_warning, match_expr=match, **kwargs
```

Comment:
yeah, let's fix these `**kwargs` in this PR

Reply:
So revert commit dfe54cd82f55f17f3c9e6e078325f306a046b93b? (I think that'd be best too).

Reply:
agreed, CC @nicoddemus